daemon/call_interfaces.c

```diff
@@ -680,6 +680,18 @@ static const char *call_offer_answer_ng(bencode_item_t *input, struct callmaster
 	if (sdp_streams(&parsed, &streams, &flags))
 		goto out;
 
+    /* Failover scenario because of timeout on offer response: siprouter tries
+     * to establish session with another rtpengine2 even though rtpengine1
+     * might have persisted part of the session. rtpengine2 deletes previous
+     * call in memory and recreates an OWN call in redis */
+	if (opmode == OP_OFFER) {
+		call = call_get(&callid, m);
+		if (call && IS_FOREIGN_CALL(call)) {
+			rwlock_unlock_w(&call->master_lock);
+			call_destroy(call);
+			obj_put(call);
+		}
```

Comment:
Good point. My primary concern is the duplicate hash lookup in the most common case. I think this can be avoided by skipping the second call_get* if the first one returned a valid call. Maybe that would be the best option.